### PR TITLE
Fix x-axis alignment on resize 

### LINKED ISSUE #43: Line chart values do not perfectly align with x-axis in smaller dimensions
image

### PR DESCRIPTION
**Pull request by t-mullen**
### Updates

Fixes #43. Data points are always aligned with the x-axis now.

Previous code assumed left margin (or more precisely, the distance from the y-axis to the first tick) was always 40, which isn't true on resize. This uses half the tick value, which is correct. Also removes an extra padding.

### Demo screenshot or recording
screen shot 2019-01-17 at 3 55 56 pm

### Review checklist (for reviewers only)

- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/src/base-axis-chart.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class BaseAxisChart extends BaseChart {
 			const chartSize = this.getChartSize();
 			const width = chartSize.width - margins.left - margins.right;
 
-			this.x = scaleBand().rangeRound([0, width]).padding(Configuration.scales.x.padding);
```
Comment (theiliad) [inline packages/core/src/base-axis-chart.ts]: This seems to be breaking the spacing in barChart

File: packages/core/src/line-chart.ts [hunk 2]
```diff
@@ -218,6 +218,11 @@ export class LineChart extends BaseAxisChart {
 		super.resizeChart();
 	}
 
+	setXScale () {
+		super.setXScale();
+		this.x.padding(0);
```
Comment (theiliad) [inline packages/core/src/line-chart.ts]: This is good, but can we add a comment to reference back to baseAxis?

Comment (theiliad) [reply packages/core/src/line-chart.ts]: And an empty line between the 2 calls, to distinguish

Comment (t-mullen) [reply packages/core/src/line-chart.ts]: What kind of comment do you mean? Just one saying that super refers to BaseAxis?

Comment (theiliad) [reply packages/core/src/line-chart.ts]: no a comment saying why we're overriding the padding, and that it is set in the `setXScale()` in BaseAxis

**Pull request merged**